### PR TITLE
Macaroon not empty check - InitWallet

### LINKED ISSUE #646: [InitWallet] macaroon is not always returned
Macaroon is not always returned in stream responses of InitWallet().

Successful call
Screenshot 2022-10-04 at 15 11 36

Unsuccessful call
Screenshot 2022-10-04 at 15 00 26

Steps to replicate:
- Run daemon (without TLS for simplicity)
```
docker run -it -d --name tdexd \
  -p 9945:9945 -p 9000:9000 \
  -v "$(pwd)/tdexd:/home/user/.tdex-daemon" \
  -e TDEX_NO_MACAROONS=false \
  -e TDEX_CONNECT_PROTO=http \
  -e TDEX_NO_OPERATOR_TLS=true \
  -e TDEX_NETWORK="regtest" \
  -e TDEX_BASE_ASSET="$TDEX_BASE_ASSET" \
  -e TDEX_LOG_LEVEL=5 \
  -e TDEX_EXPLORER_ENDPOINT="http://chopsticks-liquid:3000" \
  --network="nigiri" \
  ghcr.io/tdex-network/tdexd:latest
```

Call on Bloom
```
{
  "wallet_password": "aaaaaaaa",
  "seed_mnemonic": [
    "length", 
    "march", 
    "alarm", 
    "eagle", 
    "recycle", 
    "few", 
    "more", 
    "denial", 
    "canyon", 
    "physical", 
    "meadow", 
    "wish", 
    "ball", 
    "coil", 
    "copper", 
    "cry", 
    "honey", 
    "then", 
    "rigid", 
    "mind", 
    "misery", 
    "access", 
    "melt", 
    "obvious"
  ],
  "restore": false
}
```

Stop daemon and rerun 5 or 10 times until you encounter the bug that happens randomly.

### PR DESCRIPTION
**Pull request by sekulicd**
This checks if macaroon file is empty in InitWallet RPC call and retries reading from file until populated.

This closes #646 

@tiero @altafan please review.

### CODE REVIEW

File: internal/interfaces/grpc/handler/walletunlocker.go [hunk 1]
```diff
@@ -120,24 +120,33 @@ func (w *walletUnlockerHandler) initWallet(
...
+		if len(mac) > 0 {
+			macStr := hex.EncodeToString(mac)
+			if err := stream.Send(&daemonv1.InitWalletResponse{
+				Data:    macStr,
+				Account: -1,
+			}); err != nil {
+				return err
+			}
+		} else {
+			goto macExist
```
Comment (altafan) [inline internal/interfaces/grpc/handler/walletunlocker.go:149]:
```suggestion
		// Retry checking the admin.macaroon file exists until it's found in the datadir.
		for {
			if _, err := os.Stat(w.adminMacaroonPath); err != nil {
				if os.IsNotExist(err) {
					continue
				}
				return err
			}

		         mac, err := ioutil.ReadFile(w.adminMacaroonPath)
		         if err != nil {
			          return err
		         }
		         if len(mac) == 0 {
		                  continue
		         }
			if err := stream.Send(&daemonv1.InitWalletResponse{
				Data:    hex.EncodeToString(mac),
				Account: -1,
			}); err != nil {
				return err
			}
		}
```

Comment (altafan) [reply internal/interfaces/grpc/handler/walletunlocker.go:149]: cc @sekulicd

Comment (sekulicd) [reply internal/interfaces/grpc/handler/walletunlocker.go]: Done

**Review by tiero — APPROVED** [review]

**Review by altafan — APPROVED** [review]

**Comment by tiero** [comment]
Shall we merge?

**Pull request merged**